boards/BMS/Src/App/states/App_PreChargeState.c

```diff
@@ -20,11 +49,67 @@ static void
     PreChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
     App_AllStatesRunOnTick100Hz(state_machine);
+
```

Comment:
might be nicer to read if we have a single state variable and updated the variable within our if-else statements like so:

```
state = precharge
if(voltage not rising OR voltage rising too quick)
   state = fault
else if (precharge ok)
   state = drive/charging

// so on...

App_SetNextState(state_macihne, state)

```

Reply:
we dont want to be re-entering precharge over and over tho

Reply:
were dont have to be in precharge over and over, this is more or less initializing our state variable. @timbitss did something similar here

Reply:
Check out how the state variable is updated on his current PR `DriveStateRunOnTick100Hz`